psc/Main.hs

```diff
@@ -78,6 +78,20 @@ compile (PSCMakeOptions inputGlob inputForeignGlob outputDir opts usePrefix) = d
             hPutStrLn stderr (P.prettyPrintMultipleWarnings (P.optionsVerboseErrors opts) warnings')
           exitSuccess
 
+globWarningOnMisses :: String -> [FilePath] -> IO [FilePath]
+globWarningOnMisses fileType = go ([], [])
+  where
+  go (hits, misses) [] = do
+    unless (null misses) $ do
+      let formattedMisses = tail $ init $ show misses
```

Comment:
Also, I disagree about chopping being less nasty :) It's a small example, but it's a bad practice in general because you're forcing evaluation of the whole string.